src/postgres/tree/PostgresDatabaseTreeItem.ts

```diff
@@ -94,10 +99,52 @@ export class PostgresDatabaseTreeItem extends AzureParentTreeItem<ISubscriptionC
         credentialsTreeItem.commandArgs = [this.parent];
         return [credentialsTreeItem];
     }
+
     public async deleteTreeItemImpl(): Promise<void> {
         const client: PostgreSQLManagementClient = createAzureClient(this.root, PostgreSQLManagementClient);
         await client.databases.deleteMethod(azureUtils.getResourceGroupFromId(this.fullId), this.parent.name, this.databaseName);
     }
+
+    private async listFunctions(clientConfig: ClientConfig): Promise<IPostgresFunction[]> {
+        const client = new Client(clientConfig);
+        await client.connect();
+
+        // Adapted from https://aka.ms/AA83fg8
+        const functionsQuery: string = `select n.nspname as schema,
+            p.proname as name,
+            p.oid as oid,
+            case when l.lanname = 'internal' then p.prosrc
+                else pg_get_functiondef(p.oid)
+                end as definition
+            from pg_proc p
+            left join pg_namespace n on p.pronamespace = n.oid
+            left join pg_language l on p.prolang = l.oid
+            where n.nspname not in ('pg_catalog', 'information_schema')
+                ${this.parent.supportsStoredProcedures() ? "and p.prokind = 'f'" : '' /* Only select functions, not stored procedures */}
+            order by name;`;
+
+        const queryResult: QueryResult = await client.query(functionsQuery);
+        const rows: { schema: string, name: string, oid: number, definition: string }[] = queryResult.rows || [];
+
+        const allNames: Set<string> = new Set();
+        const duplicateNames: Set<string> = new Set();
+        for (const row of rows) {
+            if (allNames.has(row.name)) {
+                duplicateNames.add(row.name);
+            }
+
+            allNames.add(row.name);
+        }
+
+        return rows.map(row => {
+            return {
+                name: row.name,
+                oid: row.oid,
+                description: duplicateNames.has(row.name) ? row.schema : '',
```

Comment:
This method is about getting the "raw" functions themselves, not converting them into tree items. Since `description` is a treeItem-specific property, I would prefer if you passed along `isDuplicate` and `schema` on this object and defined `description` in the tree item itself. (I feel like we might need `schema` in the future anyways)

Something like
```
public get description(): string | undefined {
    return this.func.isDuplicate ? this.func.schema : undefined;
}
```